### PR TITLE
Add check premium token has :serialization feature in import/export

### LINKED ISSUE #24798: [SerDes 45] Add new premium features token feature flag for serialization features
Part of #24766

We need to add a new feature flag to signify whether it's ok to use new online single-instance SerDes features listed in #24766. `:serialization` is probably a good flag name but either way we need to coordinate with the Cloud team and have them add this to the appropriate existing tokens.

We can use [`define-premium-feature`](https://github.com/metabase/metabase/blob/master/src/metabase/public_settings/premium_features.clj#L180) to define this feature. While we're waiting for Cloud to add it we can create a placeholder function that always returns true and start using that immediately.

### PR DESCRIPTION
**Pull request by bshepherdson**
Fixes #24798
Fixes https://github.com/metabase/metabase-private/issues/92

This applies to v1 and v2 serialization.

Also adds tests for both cases to both versions: test the commands
work with :serialization enabled, and throw without it.

### CODE REVIEW

**Review by john-metabase — CHANGES_REQUESTED** [review]
Looks good, except for the audit-db tests.

**Review by john-metabase — APPROVED** [review]
Might as well approve, pending the audit-db test fix.

**Comment by john-metabase** [comment]
Do we need to check to make sure that any token that can use audit v2 has the necessary features for `v2-load` to work while initializing the audit DB?

**Comment by camsaul** [comment]
I think this fixes https://github.com/metabase/metabase-enterprise/issues/451 and https://github.com/metabase/metabase/issues/24798

File: enterprise/backend/src/metabase_enterprise/serialization/cmd.clj
Comment (noahmoss) [inline enterprise/backend/src/metabase_enterprise/serialization/cmd.clj:81]: Small suggestion, maybe this could just be the entry point for internal use cases, rather than having a separate `v2-load-no-token-check` wrapper? And then the token check could just be done unconditionally in `v2-load`

File: src/metabase/public_settings/premium_features.clj
Comment (noahmoss) [inline src/metabase/public_settings/premium_features.clj:249]: Are you re-adding this intentionally? Don't see it being used

**Review by noahmoss — APPROVED** [review]
Do you know why the `premium-features-overrides` PR broke e2e tests? That seems strange to me. Also curious why that didn't show up in CI originally and you were able to merge? Sorry if there was a discussion on this that I missed.

File: src/metabase/public_settings/premium_features.clj
Comment (bshepherdson) [reply src/metabase/public_settings/premium_features.clj:249]: Whoops, no, that should have been dropped. I guess that's a rebase ghost. I'll clear it out.

File: enterprise/backend/src/metabase_enterprise/serialization/cmd.clj
Comment (camsaul) [inline enterprise/backend/src/metabase_enterprise/serialization/cmd.clj:52]: Could we make this error message a little more user-friendly? If you try to use SerDes and see this it might not mean a lot to someone who doesn't know how our token feature flags work internally

Comment (bshepherdson) [reply enterprise/backend/src/metabase_enterprise/serialization/cmd.clj]: I've rephrased it. Happy to change it again if we can think of some better phrasing.

**Review by camsaul — APPROVED** [review]
LGTM, the only suggestion I would maybe make would be to have `:token-check?` be a keyword option, I think the fact that you have the line of code 

```clj
(v2-load-internal analytics-root-dir-resource {} #_token-check? false)
```

is a sign that the purpose of `false` isn't clear enough as a positional parameter; instead of using a comment to clarify you could just make it a keyword option e.g.

```clj
(v2-load-internal analytics-root-dir-resource {} :token-check? false)
```

Then you get that extra context for free whenever anyone uses it. Also, you won't need to pass in `true` for the normal use case, since you can make `true` the default. Plus if you want to add more options in the future you don't need to change the signature again.

But that's extremely minor and I don't want to hold up this PR on such a tiny little suggestion

**Pull request merged**
